nixos/modules/programs/sway.nix

```diff
@@ -55,16 +53,20 @@ in {
       type = types.lines;
       default = "";
       example = ''
+        # SDL:
         export SDL_VIDEODRIVER=wayland
-        # needs qt5.qtwayland in systemPackages
-        export QT_QPA_PLATFORM=wayland
+        # QT (needs qt5.qtwayland in systemPackages):
+        export QT_QPA_PLATFORM=wayland-egl
         export QT_WAYLAND_DISABLE_WINDOWDECORATION="1"
         # Fix for some Java AWT applications (e.g. Android Studio),
         # use this if they aren't displayed properly:
         export _JAVA_AWT_WM_NONREPARENTING=1
       '';
```

Comment:
Not sure how much the example should cover. Maybe the links in the description are enough? `QT_WAYLAND_DISABLE_WINDOWDECORATION` and `_JAVA_AWT_WM_NONREPARENTING` might be pretty useful though.